frame/staking/src/pallet/impls.rs

```diff
@@ -633,81 +643,136 @@ impl<T: Config> Pallet<T> {
 
 	/// Get all of the voters that are eligible for the npos election.
 	///
+	/// `voter_count` imposes a cap on the number of voters returned; care should be taken to ensure
+	/// that it is accurate.
+	///
 	/// This will use all on-chain nominators, and all the validators will inject a self vote.
 	///
 	/// ### Slashing
 	///
 	/// All nominations that have been submitted before the last non-zero slash of the validator are
 	/// auto-chilled.
-	///
-	/// Note that this is VERY expensive. Use with care.
-	pub fn get_npos_voters() -> Vec<(T::AccountId, VoteWeight, Vec<T::AccountId>)> {
-		let weight_of = Self::slashable_balance_of_fn();
-		let mut all_voters = Vec::new();
+	pub fn get_npos_voters(
+		maybe_max_len: Option<usize>,
+	) -> Vec<(T::AccountId, VoteWeight, Vec<T::AccountId>)> {
+		let weight_of = Self::weight_of_fn();
+
+		let nominator_count = CounterForNominators::<T>::get() as usize;
+		let validator_count = CounterForValidators::<T>::get() as usize;
+		let all_voter_count = validator_count.saturating_add(nominator_count);
+
+		let max_allowed_len = maybe_max_len.unwrap_or(all_voter_count).min(all_voter_count);
+		drop(all_voter_count);
```

Comment:
@kianenigma what is the signifance of calling drop on `all_voter_count`?

Reply:
henceforth in the function, `max_allowed_len` should be used as the *total*, no longer `all_voter_count`. I actually had this mistake in my code at once, and decided to drop the value to make sure it doesn't happen again.